### PR TITLE
Loader Typescript migration

### PR DESCRIPTION
**Pull request by SergeyRoyt**
https://monday.monday.com/boards/2861766393/pulses/3140977368

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,10 +20,9 @@
     "test:jest:stories": "testing=storybook TEST_END_FILES=jest-story jest",
     "test:jest:update": "npm run test:jest -- -u",
     "test:jest:coverage": "npm run test:jest -- --coverage",
-    "start": "npm run build:ts-index && npm run build-react-icons && webpack --mode production --watch --config webpack/default.config.js",
-    "build": "npm run build:exports && npm run build-react-icons && webpack --mode production --config webpack/default.config.js",
-    "build:exports": "./scripts/build-exports.js",
-    "build:ts-index": "./scripts/build-ts-index.js",
+    "start": "npm run build-react-icons && webpack --mode production --watch --config webpack/default.config.js",
+    "build": "npm run build-react-icons && webpack --mode production --config webpack/default.config.js && npm run build:esm",
```
Comment (hadasfa) [inline package.json:24]: why moving build:esm to be last? I think we moved it to be first for a reason (not remember why)

Comment (SergeyRoyt) [reply package.json:24]: Where was it? 
Haven't moved in on purpose

File: src/components/Loader/LoaderConstants.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+export enum LoaderColors {
+  PRIMARY = "var(--primary-color)",
+  SECONDARY = "var(--secondary-text-color)",
+  ON_PRIMARY = "var(--text-color-on-inverted)",
+  DARK = "var(--primary-text-color)"
+}
+
+export const LoaderSizes = {
```
Comment (hadasfa) [inline src/components/Loader/LoaderConstants.ts:8]: I think we can convert it to regular enum, no?

File: src/components/Loader/LoaderConstants.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  XS: 16,
+  SMALL: 24,
+  MEDIUM: 40,
+  LARGE: 64
+};
+
+export type LoaderSize = typeof LoaderSizes | number;
+// export type LoaderSize = typeof LoaderSizes[keyof typeof LoaderSizes];
```
Comment (hadasfa) [inline src/components/Loader/LoaderConstants.ts:16]: let's remove notes if not needed?

File: webpack/published-js-components.js [hunk 4]
```diff
@@ -51,8 +51,11 @@ const publishedJSComponents = {
   TextField: "components/TextField/TextField.jsx",
   Toast: "components/Toast/Toast.jsx",
   Tooltip: "components/Tooltip/Tooltip.jsx",
-  Loader: "components/Loader/Loader.jsx",
+  Icon: "components/Icon/Icon.jsx",
```
Comment (hadasfa) [inline webpack/published-js-components.js:54]: is it by purpose? I know some of these components already converted to ts (icon for example).
let's pull from master and mover the components you already converted to published-ts-components.js

**Review by hadasfa — CHANGES_REQUESTED** [review]
wrote few notes :)

File: webpack/published-js-components.js (hunk 4 above)
Comment (SergeyRoyt) [reply webpack/published-js-components.js:54]: Some mess happened, thanks for noticing! Fixed :)

File: src/components/Loader/Loader.tsx [hunk 5]
```diff
@@ -66,28 +74,15 @@ const Loader = forwardRef(
...
-};
+Object.assign(Loader, {
+  sizes: LoaderSizes,
+  colors: LoaderColors
+});
 
 Loader.defaultProps = {
   id: undefined,
   className: undefined,
+  svgClassName: undefined,
```
[the remark this answers is not in the document]
Comment (SergeyRoyt) [reply src/components/Loader/Loader.tsx]: Was looking at button example, while doing it) Fixed

**Pull request merged**